doc/dev/adr/0023-property-support.md

```diff
@@ -0,0 +1,312 @@
+  * For dynamically itemized properties, the keys in the `TProp` response map indicate which Properties have been read, and the corresponding values are what was read.
+
+The *observe* action is performed by issuing an 'observe' Command request.
+
+* The request payload is an instance of `TCtrl`.
+  * For statically itemized properties, the optional fields in the `TCtrl` request object are `true` for each field whose Property is to be added to the notify list.
+  * For dynamically itemized properties, the elements in the `TCtrl` request array indicate the names of the Properties to add to the notify list.
+  * The maintainer may keep a single notify list across all consumers, or it may keep a separate notify list for each consumer; in the latter case, the maintainer must be provided with a way to identify the consumer that requested the observation.
+* The response payload is an instance of `TStat`.
+  * For statically itemized properties, each optional field in the `TStat` response object indicates the status of the observe operation for the Property named by the field.
+  * For dynamically itemized properties, each value in the `TStat` response map indicates the status of the observe operation for the Property named by the key.
+  * The semantics of `TStatVal` values are determined by the service that implements the property.
+  * The maintainer's response may additionally provide status values for the observation state of properties other than those named in the 'observe' request.
+
+The *unobserve* action is performed by issuing an 'unobserve' Command request.
+
+* The request payload is an instance of `TCtrl`.
+  * For statically itemized properties, the optional fields in the `TCtrl` request object are `true` for each field whose Property is to be removed from the notify list.
+  * For dynamically itemized properties, the elements in the `TCtrl` request array indicate the names of the Properties to remove from the notify list.
+  * The maintainer may keep a single notify list across all consumers, or it may keep a separate notify list for each consumer; in the latter case, the maintainer must be provided with a way to identify the consumer that requested the observation.
+* The response payload is an instance of `TStat`.
+  * For statically itemized properties, each optional field in the `TStat` response object indicates the status of the unobserve operation for the Property named by the field.
+  * For dynamically itemized properties, each value in the `TStat` response map indicates the status of the unobserve operation for the Property named by the key.
+  * The semantics of `TStatVal` values are determined by the service that implements the property.
+  * The maintainer's response may additionally provide status values for the observation state of properties other than those named in the 'unobserve' request.
+
+The *notify* action is performed by sending a Telemetry with a payload that is an instance of `TProp`.
```

Comment:
If we support shared notification lists (where one consumer can affect the notifications that they and other consumers will receive), I think it is important to have a notification for when something becomes unobserved (and potentially for when something becomes observed as well for symmetry, although I don't think the ramifications of not having that one are as large)